src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,504 @@
+        {
+            get
+            {
+                return _resourceCulture;
+            }
+            set
+            {
+                _resourceCulture = value;
+            }
+        }
+
+        private static string GetResourceString(string resourceName)
+        {
+            string value = ResourceManager.GetString(resourceName, Culture);
+            if (string.IsNullOrEmpty(value))
+            {
+                value = string.Format(CultureInfo.InvariantCulture, "Unknown resource: {0}", resourceName);
+                Diagnostics.Assert(false, value);
+            }
+            return value;
+        }
+
+        #endregion resource manager
+
+        /// <summary>
+        /// Gets the EventMessage for a given event.
+        /// </summary>
+        /// <param name="sb">The StringBuilder to append.</param>
+        /// <param name="eventId">The id of the event to retrieve.</param>
+        /// <param name="args">An array of zero or more payload objects</param>
+        private static void GetEventMessage(StringBuilder sb, PSEventId eventId, params object[] args )
+        {
+            EventResource resource = EventResource.GetMessage((int) eventId);
+
+            if (resource == null)
+            {
+                resource = EventResource.GetMissingEventMessage();
+                string resourceValue = GetResourceString(resource.Name);
+                sb.AppendFormat(CultureInfo.InvariantCulture, resourceValue, (int) eventId);
+
+                // If an event id was specified that is not found in the event resource lookup table,
+                // use a placeholder message that includes the event id.
+                Diagnostics.Assert(false, sb.ToString());
+            }
+            else
+            {
+                string resourceValue = GetResourceString(resource.Name);
+                if (resource.ParameterCount > 0)
+                {
+                    sb.AppendFormat(resourceValue, args);
+                }
+                else
+                {
+                    sb.Append(resourceValue);
+                }
+            }
+        }
+
+        #region logging
+
+        // maps a LogLevel to an associated SysLogPriority
+        static NativeMethods.SysLogPriority[] _levels =
+        {
+            NativeMethods.SysLogPriority.Info,
+            NativeMethods.SysLogPriority.Critical,
+            NativeMethods.SysLogPriority.Error,
+            NativeMethods.SysLogPriority.Warning,
+            NativeMethods.SysLogPriority.Info,
+            NativeMethods.SysLogPriority.Info
+        };
+
+        /// <summary>
+        /// Logs a activity transfer
+        /// </summary>
+        /// <param name="parentActivityId">The </param>
+        public void LogTransfer(Guid parentActivityId)
+        {
+            // NOTE: always log
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            string message = string.Format
```

Comment:
Formatting - `(` etc. should not be on a line by itself.

Reply:
Fixed